src/compiler/scala/tools/cmd/CommandLineParser.scala

```diff
@@ -1,82 +1,103 @@
 /* NEST (New Scala Test)
- * Copyright 2007-2013 LAMP/EPFL
+ * Copyright 2007-2018 LAMP/EPFL
  * @author Paul Phillips
  */
-
-package scala.tools
-package cmd
+package scala.tools.cmd
 
 import scala.annotation.tailrec
 
-/** A simple (overly so) command line parser.
- *  !!! This needs a thorough test suite to make sure quoting is
- *  done correctly and portably.
+/** A simple enough command line parser.
  */
 object CommandLineParser {
-  // splits a string into a quoted prefix and the rest of the string,
-  // taking escaping into account (using \)
-  // `"abc"def` will match as `DoubleQuoted(abc, def)`
-  private class QuotedExtractor(quote: Char) {
-    def unapply(in: String): Option[(String, String)] = {
-      val del = quote.toString
-      if (in startsWith del) {
-        var escaped = false
-        val (quoted, next) = (in substring 1) span {
-          case `quote` if !escaped => false
-          case '\\'    if !escaped => escaped = true; true
-          case _                   => escaped = false; true
-        }
-        // the only way to get out of the above loop is with an empty next or !escaped
-        // require(next.isEmpty || !escaped)
-        if (next startsWith del) Some((quoted, next substring 1))
-        else None
-      } else None
+  private final val DQ = '"'
+  private final val SQ = '\''
+
+  /** Split the line into tokens separated by whitespace or quotes.
+   *
+   *  @return either an error message or reverse list of tokens
+   */
+  private def tokens(in: String) = {
+    import Character.isWhitespace
+    import java.lang.{StringBuilder => Builder}
+    import collection.mutable.ArrayBuffer
+
+    var accum: List[String] = Nil
```

Comment:
Isn't a `ListBuffer` faster than reversing a prepend-built List?

https://stackoverflow.com/a/1241394/463761 seems to indicate it would be constant-time appends + constant-time convert to List, rather than constant-time prepends and linear-time reverse.

Reply:
Thanks, I'll add that to the collections FAQ. I inherited it from the previous code; but I changed that ArrayBuffer from a ListBuffer, without much analysis. This was an exercise in using jmh, but I failed even in that goal.